src/dfastbe/bank_erosion/data_models/inputs.py

```diff
@@ -268,6 +276,234 @@ def _read_river_axis(self) -> LineString:
+            Reed values for each bank.
+    """
+    Parameters: ClassVar["Parameters"] = namedtuple(
+        "Parameters", "name default valid onefile positive ext"
+    )
+    config_file: ConfigFile
+    velocity: List[np.ndarray]
+    number: List[np.ndarray]
+    num_waves: List[np.ndarray]
+    draught: List[np.ndarray]
+    type: List[np.ndarray]
+    slope: List[np.ndarray]
+    reed: List[np.ndarray]
+
+    @classmethod
+    def get_ship_data(
+        cls, num_stations_per_bank: List[int], config_file: ConfigFile
+    ) -> "ShipsParameters":
+        """Get ship parameters from the configuration file.
+
+        Args:
+            num_stations_per_bank (List[int]):
+                The number of stations per bank.
+            config_file (ConfigFile):
+                Configuration file containing parameters.
+
+        Returns:
+            ShipsParameters: An instance of ShipsParameters with parameters read from the config file.
+        """
+
+        param_defs = cls._get_initial_parameter_definitions()
+        param_resolved = {}
+
+        # Retrieve parameter values
+        for param in param_defs:
+            param_resolved[f"{param.name.lower()}"] = cls._get_param(
+                config_file,
+                param.name,
+                num_stations_per_bank,
+                default=param.default,
+                valid=param.valid,
+                onefile=param.onefile,
+                positive=param.positive,
+                ext=param.ext,
+            )
+
+        param_dict = {
+            "velocity": param_resolved["vship"],
+            "number": param_resolved["nship"],
+            "num_waves": param_resolved["nwave"],
+            "draught": param_resolved["draught"],
+            "type": param_resolved["shiptype"],
+            "slope": param_resolved["slope"],
+            "reed": param_resolved["reed"],
+        }
+
+        return cls(config_file, **param_dict)
+
+    @staticmethod
+    def _get_param(
+        config_file: ConfigFile, name: str, num_stations_per_bank, iq_str="", **kwargs
+    ):
+        """Get a parameter from the configuration file.
+
+        Args:
+            config_file (ConfigFile):
+                Configuration file containing parameters.
+            name (str):
+                Name of the parameter to retrieve.
+            num_stations_per_bank (List[int]):
+                The number of stations per bank.
+            iq_str (str, optional):
+                String to append to the parameter name for specific levels.
+            **kwargs:
+                Additional keyword arguments for parameter retrieval.
+        Returns:
+            np.ndarray: The parameter value as a NumPy array.
+        """
+        return config_file.get_parameter(
+            "Erosion",
+            f"{name}{iq_str}",
+            num_stations_per_bank,
+            **kwargs,
+        )
+
+    @classmethod
+    def _get_initial_parameter_definitions(cls) -> List[Parameters]:
```

Comment:
why there is `_get_initial_parameter_definitions` and `_get_discharge_parameter_definitions` ?